power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -510,6 +513,27 @@ template <dataset_type_tag dataset_type_> class Dataset {
         return result;
     }
 
+    // get slice dataset from batch
+    Dataset get_slice_scenario(Idx begin, Idx end) const
```

Comment:
really useful :+1: If we add an additional `increment` here, we can actually also use this in the dispatching of multithreaded batch calculations